File: src/main/java/com/ebayfff/service/WishListService.java
package com.ebayfff.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.ebayfff.domain.WishList;
import com.ebayfff.dto.WishListDTO;
import com.ebayfff.dto.WishListSearchDTO;
import com.ebayfff.dto.WishListPageDTO;
import com.ebayfff.dto.WishListConvertCriteriaDTO;
import com.ebayfff.service.GenericService;
import com.ebayfff.dto.common.RequestDTO;
import com.ebayfff.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface WishListService extends GenericService<WishList, Integer> {

	List<WishList> findAll();

	ResultDTO addWishList(WishListDTO wishListDTO, RequestDTO requestDTO);

	ResultDTO updateWishList(WishListDTO wishListDTO, RequestDTO requestDTO);

    Page<WishList> getAllWishLists(Pageable pageable);

    Page<WishList> getAllWishLists(Specification<WishList> spec, Pageable pageable);

	ResponseEntity<WishListPageDTO> getWishLists(WishListSearchDTO wishListSearchDTO);
	
	List<WishListDTO> convertWishListsToWishListDTOs(List<WishList> wishLists, WishListConvertCriteriaDTO convertCriteria);

	WishListDTO getWishListDTOById(Integer wishListId);







}

File: ebayfff_ui/src/services/WishListService.js
import http from "../http-common";

class WishListService {
  getAllWishLists(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/wishList/wishLists`, searchDTO);
  }

  get(wishListId) {
    return this.getRequest(`/wishList/${wishListId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/wishList?field=${matchData}`, null);
  }

  addWishList(data) {
    return http.post("/wishList/addWishList", data);
  }

  update(data) {
  	return http.post("/wishList/updateWishList", data);
  }
  
  uploadImage(data,wishListId) {
  	return http.postForm("/wishList/uploadImage/"+wishListId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new WishListService();
